exercises/error_handling/errors4.rs

```diff
@@ -16,8 +14,7 @@ enum CreationError {
 
 impl PositiveNonzeroInteger {
     fn new(value: i64) -> Result<PositiveNonzeroInteger, CreationError> {
-        // Hmm... Why is this always returning an Ok value?
-        Ok(PositiveNonzeroInteger(value as u64))
+        if value == 0 { Err(CreationError::Zero) } else if value < 0 { Err(CreationError::Negative) } else { Ok(PositiveNonzeroInteger(value as u64)) }
```

Comment:
You **MUST** use a formatter.
If you're using VSCode please [setup `rust-analyzer`](https://rust-analyzer.github.io/manual.html#vs-code) extension and configure it to [auto-format your code on save](https://stackoverflow.com/a/39973431)!

Reply:
Oh that's great tip